### DISCUSSION TITLE
Icon Design Guidelines - Current & Future

### DISCUSSION THREAD

**Discussion by patepelo** [organicmaps/organicmaps#5294] [Ideas]
Hello, 

The purpose of this thread is to understand the current iconography and study possible changes. Based from the wiki: https://github.com/organicmaps/organicmaps/wiki/Icons

### Current Icons seem to follow this logic in general:

There are mainly circled, square and open icons (like sports).

Circled icons:

The tendency is that circled are used in general for ammenities administrated by someone present:
Like a shop:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/fddbe9a6-8b26-4c07-b8f4-31aaba3cc017)
Or a bar:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/2d25d54d-3950-4c70-a9b1-ffdb79972327)
Or a hostel:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/264eabfc-adfc-4468-a349-80530e2e6b8f)
Or a church:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/65ed4336-9abe-497e-91b8-0d748abe8450)
Fuel station:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/a1c52796-3e29-4671-90b8-d9470b7e6ccd)

Colors are somehow clear according to the wiki.

Square icons:
The tendency is for transport uses, including parking. 
![image](https://github.com/organicmaps/organicmaps/assets/23003127/1a930772-b67a-4261-bfec-f17ef6112351)
![image](https://github.com/organicmaps/organicmaps/assets/23003127/1f52a78b-219e-4b12-9514-03a05afa9874)
![image](https://github.com/organicmaps/organicmaps/assets/23003127/4e2a4263-1493-43ca-84b6-0d48b9518b00)

Open icons:
The tendency is for objects that are permanently open and unattended.

Like table tennis open tables:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/d60ab705-9fdf-4d59-a603-36d9b2be9e9b)
Or viewpoints:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/10894d5f-aa5d-4ba5-82ad-5d4e726ecf15)
Or fuel vending machines:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/a82c6ee0-80b0-424f-8cef-ddd252897831)
Or bicycle repair stations:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/e3d6f8cc-22c7-4444-891a-5098ab6c1a53)
Picnic tables:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/f1dd61ba-41ad-410a-b10e-081a976ae7ff)

### There are some cases where this criteria isn't followed, mainly in the sports category. Here some examples:

Yoga Studios: 
![image](https://github.com/organicmaps/organicmaps/assets/23003127/5cf0555e-eb47-4153-9aa9-731371aba30c)

Fitness Centres:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/d6f9a977-ba1b-4989-8a23-7e9e6ddc791e)

Climbing Gyms:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/3de5e809-6782-4a57-86fe-43498f0a2fd2)

- I start the discussion with these. Should we try to update them to follow the general guidelines?

**Comment by patepelo** [comment]
Other things to consider are public squares, maybe they could have their own icon (open one) similar to parks guideline: 
![image](https://github.com/organicmaps/organicmaps/assets/23003127/3cf3b381-04d4-4e3a-96ca-ee81a3c0d75f)
 
This would differentiate them from the Attractions Icon 
![image](https://github.com/organicmaps/organicmaps/assets/23003127/97cb7e2f-b631-4993-8f6d-756e1ac733dd)

**Reply by biodranik** [reply]
1. Are public squares rendered as attractions now?
2. Historically, some icons in brown circles were related to "touristic objects". Are attractions businesses? Is there a better UX idea about differentiating touristic objects visually?

**Reply by Mr-Mime** [reply]
1. From what I see yes, see this example:

Node on osm: https://www.openstreetmap.org/way/572780583
In OMaps: 
<img src="https://github.com/organicmaps/organicmaps/assets/37423773/d01863d6-f353-494d-b0ed-cb747aca8132" width="400px">

2. I don't think that attractions are businesses. While some sell tickets and therefore have a person, it still is not a business in my understanding.

**Reply by hb0nd** [reply]
(we decided against an icon for `place=square` in https://github.com/organicmaps/organicmaps/issues/6013 👍️)

**Comment by patepelo** [comment]
These are some first drafts on the Yoga Studio and Fitness Centres:

![image](https://github.com/organicmaps/organicmaps/assets/23003127/7e7d1a9f-f764-4b4f-bc0c-1db6bcd42afc)

**Reply by biodranik** [reply]
You got it right. These should be in circles too. Circles mean "a business, place, or office operated by humans". Colors for different groups are discussable (e.g. sport-related businesses, food/drink, etc.)

What about dark mode variants? Also it would be great to see these icons on the map screenshot.

**Comment by patepelo** [comment]
Let me invite you from previous Icons Discussions, @pastk @RedAuburn @biodranik @Mr-Mime @AlertSubject @TheAdventurer64

**Comment by patepelo** [comment]
With the latest changes we've been doing about adding or removing background circles I've been thinking there might be icons that are better the way they are right now, these would be the exceptions:

Toilettes:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/24adaaed-7504-4c70-a87d-953f5cd8df18)

Mine / Quarry:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/fb0c80a4-cafb-444a-a3af-d5829b08fb71)

ATM:
![image](https://github.com/organicmaps/organicmaps/assets/23003127/2749b11f-2c7f-43e9-b458-8575feb51746)

What are your thoughts? Any other worth mentioning?

**Reply by biodranik** [reply]
Do they look bad without circles? Let's try it.

**Comment by patepelo** [comment]
This is just a preview and thus not very acccurate.

![image](https://github.com/organicmaps/organicmaps/assets/23003127/d49ddf31-7308-4543-a61b-fe3ecb08cf91)
![image](https://github.com/organicmaps/organicmaps/assets/23003127/e2b8c07c-820f-4f44-b794-253018befa83)

**Comment by pastk** [comment]
> 
> The tendency is that circled are used in general for ammenities administrated by someone present:

Let's discuss it!

What value for a user this circled / not-circled distinction bring?

Does it really matter whether a POI is "administered" / is a business?

What matters a lot is that adding a circle makes an icon much more pronounced / visible.

So logically it should be more about important/big vs minor/niche?

E.g. for tourists viewpoints and  toilets are important POIs (but not circled / less visible in the current paradigm) whereas many niche shops are not (but they're circled / more visible).

**Reply by biodranik** [reply]
The best way to search for a toilet is to use the search. Distracting toilet circles are not needed if you don't want to the toilet yet.

**Reply by pastk** [reply]
But the same applies to almost any other POI :) How does it justify using or not using circled icons?

**Reply by patepelo** [reply]
As a user, many times I go and open OM,look for toilets just exploring the map around searching for the icon. I think it's useful, just maybe only in the closer zoom levels.

**Reply by pastk** [reply]
Some time before we had most icons circled, except very minor / micro-mapped features like trash cans, benches, fire hydrants, signposts etc. which were without a circle. They were kind of more considered "a part of the scenery / relief" than a POI :)

IMO it made more sense than the current approach - why did we change it at all? What was the logic behind it?

**Reply by pastk** [reply]
so what do you think about 
1. accepting that adding a circle makes a POI stand out a lot against the background
2. dropping the "circles are for businesses / human-attended places" concept
3. using circles to distinguish between major and minor (micro-mapped?) POIs

e.g. by this logic viewpoints and picnic sites should be circled (but not picnic tables which is a micro-mapping)
and industrial factories (man_made=works) should be de-circled

**Reply by pastk** [reply]
@organicmaps/styles 👆

**Reply by hb0nd** [reply]
Generally i agree.

however, I think it'd be better to decouple the icons from the colours/circles/outlines first. As updating every icon would take probably the same amount of time as just implementing this, and then we can easily compare different styles.

https://github.com/organicmaps/organicmaps/issues/2026

**Reply by patepelo** [reply]
>2. dropping the "circles are for businesses / human-attended places" concept

I've found it very intuitive so far and like that concept but do agree that can have flexibility for circling other objects,

>e.g. by this logic viewpoints and picnic sites should be circled (but not picnic tables which is a micro-mapping)
and industrial factories (man_made=works) should be de-circled

What POIs would you like to have circled besides viewpoints and picnic sites? In these two examples, I like viewpoints to be more discreet, due to many are very subjective and not very important to find quickly. And picnic sites I'd agree that could be better circled IMO (as they group objects or represent an area rather than a single micro-mapped object).

**Reply by pastk** [reply]
I understand that viewpoints mapping could be subjective (similar to e.g. tourism=attraction), but still its an important POI regardless whether its attended or not.

Some other POIs to consider circling: fort & castles & city gates (but not ruins), public bookcase, lighthouse, windmill, climbing sites.

**Reply by patepelo** [reply]
Fort and castles, I consider great to circle.
Public book cases, not IMO by a similar principle of small unattended.
Lighthouse, I'm not sure about that. 
What kind of Windmill are you referring to? Historical? How are they currently represented?
And climbing sites, do you mean climbing gyms or outdoor features?

>dropping the "circles are for businesses / human-attended places" concept
Besides the logic so far, in general, my intuition tells me that a circled POI is something you'd expect to click on and find a place page full of information like opening hours, website, social networks, etc.

**Comment by pastk** [comment]
Related to https://github.com/organicmaps/organicmaps/issues/2205

**Comment by pastk** [comment]
Sometimes we have an issue with "small" feature icons not visible against a particular background, as they don't have any outline.
E.g. a gate on the grey highway=track.

Obstacles (gates, cycle barriers, etc.) are quite important when it comes to planning a route! We should consider adding a contract (white) outline to them just like we do for fords and named nature reserves.

**Reply by patepelo** [reply]
Seems like a good idea we could test.

Maybe if there was a form of highlighting for example passing gates once they're in the route, like we do with speed radars (while car navigation). Tell me if it's worth opening a specific issue.

![image](https://github.com/organicmaps/organicmaps/assets/23003127/5534ad7b-87c3-4bf8-a81d-7323d8021fc3)

**Reply by pastk** [reply]
yeap it would be great to have a warning (maybe there is an issue already?)

**Comment by pastk** [comment]
Fire hydrants are red bacause they're `emergency=fire_hydrants` and we color red all the emergency features.

However it makes fire hydrants too prominent on the map, so we had to reduce their visibility to z19-. At the same time benches and trash bins are z18-!
Maybe it'd be better to make fire hydrants grey but z18-?

**Reply by hb0nd** [reply]
imo this is fine, fire hydrants are a pretty niche use-case

**Reply by Outlet2048** [reply]
> However it makes fire hydrants too prominent on the map, so we had to reduce their visibility to z19-. At the same time benches and trash bins are z18-!
> Maybe it'd be better to make fire hydrants grey but z18-?


Benches and trash bins are faaaar more useful than any fire hydrant would ever be to me 
They could disappear over night and in my use case I wouldn't even notice

**Reply by biodranik** [reply]
Fire hydrants were added by firefighters' request. We can show them only on Z20 @pastk. They're not needed for normal users.
